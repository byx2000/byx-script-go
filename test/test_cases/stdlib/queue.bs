import queue

var q = Queue()
q.enQueue(1)
q.enQueue(2)
q.enQueue(3)
println(q.size(), q.isEmpty())

println(q.deQueue())
println(q.deQueue())
println(q.size(), q.isEmpty())

q.enQueue(4)
q.enQueue(5)
q.enQueue(6)
println(q.size())
println(q.front())
println(q.tail())

println(q.deQueue())
println(q.deQueue())
println(q.deQueue())
q.deQueue()
println(q.size(), q.isEmpty())
