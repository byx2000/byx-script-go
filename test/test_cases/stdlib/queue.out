3 false
1
2
1 false
4
3
6
3
4
5
0 true